libraries/Microsoft.Bot.Builder/BotState.cs

```diff
@@ -112,11 +112,8 @@ public Task ClearStateAsync(ITurnContext turnContext, CancellationToken cancella
                 throw new ArgumentNullException(nameof(turnContext));
             }
 
-            var cachedState = turnContext.TurnState.Get<CachedBotState>(_contextServiceKey);
-            if (cachedState != null)
-            {
-                turnContext.TurnState[_contextServiceKey] = new CachedBotState();
-            }
+            // Explicitly setting the hash will mean IsChanged is always true. And that will force a Save.
```

Comment:
The documentation on this method needs to be rewritten to reflect that this is now going to actually impact persisted state. (NOTE: GitHub doesn't let me comment on unedited lines or I would have put this on 103.)

Reply:
So, to be specific, I think "default form" is just confusing. I think the doc should read:

> Clears any state currently stored in this state scope.

And then I think we should add a `<remarks>` that includes a `NOTE:` that `SaveChangesAsync` must be called in order for the cleared state to be persisted to the underlying store.

Reply:
added unit test